bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java

```diff
@@ -45,20 +57,25 @@
  * @author Chris Graham - Initial contribution
  * @author Florian Schmidt - Allow https URLs and basic auth
  */
+@NonNullByDefault
 class OpenSprinklerHttpApiV100 implements OpenSprinklerApi {
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
     protected final String hostname;
     protected final int port;
-    protected final String password;
```

Comment:
Question: Why is this field not final anymore? And second: I might have missed something, but do we actually need this field anywhere? :D

Reply:
Yes I feel the field is needed as I had an issue where the password got hashed twice in a row which broke the binding until the binding was restarted. I forget the details but it may have been a simple edit of a parameter that caused this to occur. So now the config.password always holds the unhashed password, but the local password variable holds the hashed version, which is always sourced from config.password....
Making it no longer final reduced the lines of code.